introduction/edge-functions.mdx

```diff
@@ -0,0 +1,96 @@
+# Edge functions
+Edge Functions are server-side functions that run directly within your database environment. Edge functions in SQLite Cloud ensure maximum performance and minimal latency by running functions on the same node as your database.
+
+You can write edge functions directly in the SQLite Cloud dashboard using JavaScript, TypeScript, or SQL. Importing modules is not currently supported.
+
+Edge functions can be called remotely over HTTP or Websockets via API, or triggered by database events via SQLite Cloud [Webhooks](https://docs.sqlitecloud.io/docs/introduction/webhooks). Each function runs in an isolated environment using the Bun runtime. 
+
+## Getting Started
+1. Navigate to the Edge Functions page from your dashboard.
+1. Under "Development", select "Edge Function", then click the "Create" button.
+1. Write and test your function.
+    1. Select the database you want to access and an API key if necessary
+    2. When you're finished, click deploy.
+
+#### Note:
+Functions should return a JSON-serializable object with a data field:
+```JavaScript
+return {
+  data: {
+    // your return object
+  }
+}
+```
+
+## Guides
+### Interacting with your Database
+Use the global `connection` object to access and manipulate your database.
+```JavaScript
+const customers = await connection.sql`SELECT * FROM customers;`
+return {
+  data: customers
+}
+```
+Select the database you would like to access from the "Database" dropdown, or select the database you want to use in your SQL query with the [USE command](https://docs.sqlitecloud.io/docs/commands/use-database).
+```JavaScript
+const customers = await connection.sql`USE DATABASE chinook.sqlite; SELECT * FROM customers;`;
+return {
+  data: customers
+}
+```
+
+### Storing and Accessing Environment Variables
+Environment variables can be stored in either the server settings file, database settings file, or stored directly in your database. 
+
+Use the following commands to set and read values in your server settings file:
+* [SET KEY](https://docs.sqlitecloud.io/docs/commands/set-key) <keyname> TO <keyvalue>
+* [GET KEY](https://docs.sqlitecloud.io/docs/commands/get-key) <keyname>
+* [REMOVE KEY](https://docs.sqlitecloud.io/docs/commands/remove-key) <keyname>
+* [LIST KEYS](https://docs.sqlitecloud.io/docs/commands/list-keys)
+
+Use the following commands to set and read values in your database-specific settings file:
+* [SET DATABASE](https://docs.sqlitecloud.io/docs/commands/set-database) <database_name> KEY <keyname> TO <keyvalue> 
+* [GET DATABASE](https://docs.sqlitecloud.io/docs/commands/get-database-key) <database_name> KEY <keyname> 
+* [REMOVE DATABASE](https://docs.sqlitecloud.io/docs/commands/remove-database-key) <database_name> KEY <keyname>
+* LIST DATABASE <database_name> KEYS
+
+### Handling Errors
+In case of error we return an HTTP error code and a JSON with the error message. Manually throwing an error in your code results in a 500 response.
```

Comment:
The user has two options:
1. Manually throwing an error that results in a 500 response
2. Return an error obj with custom status. The 
```
return {
	error: {
		/** Error status as http code */
		status: string
		title?: string
		detail?: string
	}
}

```